src/routes/registration.ts

```diff
@@ -45,8 +51,27 @@ export const createRegistrationRoute: RouteCreator =
       .then(({ data: flow }) => {
         // Render the data using a view (e.g. Jade Template):
         res.render("registration", {
-          ...flow,
-          signInUrl: initLoginUrl,
+          nodes: flow.ui.nodes,
+          webAuthnHandler: filterNodesByGroups({
+            nodes: flow.ui.nodes,
+            groups: ["webauthn"],
+            attributes: ["button"],
+            withoutDefaultAttributes: true,
+            withoutDefaultGroup: true,
+          })
+            .filter(({ attributes }) => isUiNodeInputAttributes(attributes))
+            .map(({ attributes }) => {
+              return (attributes as UiNodeInputAttributes).onclick
+            })
+            .filter((c) => c !== undefined),
+          card: UserAuthCard({
+            title: "Register an account",
+            flow: flow as SelfServiceFlow,
```

Comment:
I think we should either add a check that `flow` actually satisfies the type or adjust it in elements. Either way, this typecast is not great, IMO.

Reply:
This is casting it to ory elements type which is the same underlying type given by the SDK.

So e.g. here flow can be SelfServiceRegistrationFlow but ory/elements expects SelfServiceFlow which is a type defined as 
```ts
type SelfServiceFlow = SelfServiceRegistrationFlow | SelfServiceLoginFlow ...
```

I don't see this breaking somewhere. We could move this type to the SDK directly at some point and have it used everywhere instead of this workaround.

Reply:
Hmm, I see. 

>I don't see this breaking somewhere. 

Famous last words ;)

> `type SelfServiceFlow = SelfServiceRegistrationFlow | SelfServiceLoginFlow ...`

In that case, the typecast should not be necessary at all, right?

Reply:
you are correct :)
I think i added this long ago when i was still busy with elements...